### DISCUSSION TITLE
Mysql error while upgrading from 16.04 to 18.04 / SO2.3.x

### DISCUSSION THREAD

**Discussion by femaven** [Security-Onion-Solutions/securityonion#1968] [Unsupported Versions] [answered]
I'm getting the following error when upgrading Security Onion from 16.04 to 18.04 / SO2.3.x

- Exact version. ex. 2.3.2
- Install source. Did you install from our Security Onion ISO image or did you perform a network installation? network install from 16.04
- If network install, did you install from CentOS 7 or Ubuntu 18.04? Ubuntu 18.04
- Install type. ex. eval, standalone, forward node, search node, etc - Eval 
- Does so-status show all services running? The only service that's running is docker-registry. 
- Do you get any failures when you run salt-call state.highstate from the node? Yes
Local:
    Data failed to compile:
----------
    The function "state.highstate" is running as PID 11176 and was started at 2020, Nov 17 20:11:03.450497 with jid 2020111720110345049

I'm seeing two issues:
1. salt-master is not running - I've tried installing salt-master 
2. Error response from daemon: No such container: so-mysql

- 3% - GENERATING CA AND CHECKING IN
----
Building Certificate Authority
local:
    Data failed to compile:
----------
    The function "state.highstate" is running as PID 11176 and was started at 2020, Nov 17 20:11:03.450497 with jid 20201117201103450497
 *** Restarting Salt to fix any SSL errors. ***
Stopping service salt-master
Checking service salt-master status
salt-master is not running
Stopping service salt-minion
Checking service salt-minion status
salt-minion is not running
Starting service salt-master
Checking service salt-master status
salt-master is running
Starting service salt-minion
Checking service salt-minion status
salt-minion is running
Checking if we can talk to the salt master
salt minion cannot talk to salt master
......
salt minion could not talk to salt master after 30 attempts, exiting
Errors detected during setup; skipping post-setup steps to allow for analysis of failures.
Installer removing the following files:
/root/installtmp:
total 4
drwxr-xr-x 3 root root 4096 Nov 17 20:12 pillar

/root/installtmp/pillar:
total 4
drwxr-xr-x 2 root root 4096 Nov 17 20:15 minions

/root/installtmp/pillar/minions:
total 4
-rw-r--r-- 1 root root 1048 Nov 17 20:17 securityonion_eval.sls
=========================================================================
Stopping mysql...
=========================================================================
Error response from daemon: No such container: so-mysql
Error: No such container: so-mysql

**Comment by Courts3003** [comment]
I am seeing exactly the same error (failure to finish installation and accompanying messages) when attempting to install a Security Onion 2.3.2 SENSOR for the distributed configuration. The "Distributed" MANAGER installs without issues. Here are the details that you asked for:

- Exact version; 2.3.2 installed from latest ISO with CentOS 7 64-bit VMware VM.
- Install source; Security Onion 2.3.2 ISO image
- If network install, did you install from CentOS 7; I chose CentOS 7 64-bit in line with installation instructions. This is a clean install of a distributed configuration.
- Install type.    SENSOR (2.3.2 ISO option).
- Does so-status show all services running?
 
OUTPUT
Checking Docker status
      Docker .. [ok]

Checking container statuses
      so-filebeat .. [ok]
      so-sensoroni .. [ok]
      so-steno   ..[ok]
      so-strelka-coordinator ..[ok]
      so-strelka-filestream   ..[ok]
      so-strelka-frontend     ..[ok]
      so-strelka-gatekeeper ..[ok]
      so-strelka-manager    ..[ok]
      so-suricata          ..[ok]
      so-telegraf           ..[ok]
       so-zeek               ..[ok]

Do you get any failures when you run salt-call state.highstate from the node?     It runs successfully; Here is a summary
Summary for local
--------------------
Succeeded:180 (changed=0)
Failed:            0
---------------------

SUMMARY
- The installation script for SENSOR fails to complete! (I have tried it 3 times now) - "no such container: so-mysql" 
- It further says "review the /root/so-setup.log" 
- "so-status"  and "salt-call" indicate a system without issues.

WAY FORWARD
- Are you able to re-produce the same error from install using 2.3.2 ISO?

**Comment by Courts3003** [comment]
ADDITIONAL UPDATE

- Despite showing an "ok" status. Zeek does not run properly, because it requires so-mysql and this is not installed (or at least does not run) by the ISO installer for SENSORS currently.
- So effectively no logs are collected if you depend on Zeek as I do.
- so-mysql-start or  restart both fail on the SENSOR.
- The MANAGER shows no alerts although it shows connection to the SENSOR.

Hope that helps resolve the issue.

**Reply by TOoSmOotH** [reply]
Zeek does not require mysql. Logs are written to disk and picked up by filebeat. The sensor will not allow you to run mysql.

**Comment by Courts3003** [comment]
Thanks for coming back Mike.

When I run so-zeek-logs on my broken SENSOR (it did not fully install), it displays the Zeek Alerts Subscription menu then prints 

/sbin/so-zeek-logs: line 8: /opt/so/saltstack/local/pillar/zeeklogs.sls: No such file or directory

multiple (more than 20) times.

**Comment by femaven** [comment]
I think what @Courts3003 and I are experiencing are 2 different issues.  As I noted in my initial post, this is an upgrade from a 16.04 eval box. The dist upgrade was successful but now I'm having issues getting SecurityOnion to work. When I run so-status I get the following, as noted in my initial post most of the services are failing and mysql is noted as missing. 

root@securityonion:/home/amit/securityonion# so-status

Checking Docker status

    Docker ------------------------------------------------------------------------ [ OK ]   

Checking container statuses

    so-cortex ------------------------------------------------------------------ [ ERROR ]   
    so-curator ----------------------------------------------------------------- [ ERROR ]   
    so-dockerregistry ------------------------------------------------------------- [ OK ]   
    so-elastalert -------------------------------------------------------------- [ ERROR ]   
    so-elasticsearch ----------------------------------------------------------- [ ERROR ]   
    so-filebeat ---------------------------------------------------------------- [ ERROR ]   
    so-fleet ------------------------------------------------------------------- [ ERROR ]   
    so-grafana ----------------------------------------------------------------- [ ERROR ]   
    so-idstools ---------------------------------------------------------------- [ ERROR ]   
    so-influxdb ---------------------------------------------------------------- [ ERROR ]   
    so-kibana ------------------------------------------------------------------ [ ERROR ]   
    so-kratos ------------------------------------------------------------------ [ ERROR ]   
    so-mysql ----------------------------------------------------------------- [ MISSING ]   
    so-nginx ------------------------------------------------------------------- [ ERROR ]   
    so-playbook ---------------------------------------------------------------- [ ERROR ]   
    so-redis ------------------------------------------------------------------- [ ERROR ]   
    so-sensoroni --------------------------------------------------------------- [ ERROR ]   
    so-soc --------------------------------------------------------------------- [ ERROR ]   
    so-soctopus ---------------------------------------------------------------- [ ERROR ]   
    so-steno ------------------------------------------------------------------- [ ERROR ]   
    so-strelka-coordinator ----------------------------------------------------- [ ERROR ]   
    so-strelka-filestream ------------------------------------------------------ [ ERROR ]   
    so-strelka-frontend -------------------------------------------------------- [ ERROR ]   
    so-strelka-gatekeeper ------------------------------------------------------ [ ERROR ]   
    so-strelka-manager --------------------------------------------------------- [ ERROR ]   
    so-suricata ---------------------------------------------------------------- [ ERROR ]   
    so-telegraf ---------------------------------------------------------------- [ ERROR ]   
    so-thehive ----------------------------------------------------------------- [ ERROR ]   
    so-thehive-es -------------------------------------------------------------- [ ERROR ]   
    so-wazuh ------------------------------------------------------------------- [ ERROR ]   
    so-zeek -------------------------------------------------------------------- [ ERROR ]

**Comment by femaven** [comment]
when I do a git pull to pull 2.3.10

I get the following results in my sosetup.log
uilding Certificate Authority
local:
    Data failed to compile:
----------
    The function "state.highstate" is running as PID 11176 and was started at 2020, Nov 17 20:11:03.450497 with jid 20201117201103450497
 *** Restarting Salt to fix any SSL errors. ***
Stopping service salt-master
Checking service salt-master status
salt-master is not running
Stopping service salt-minion
Checking service salt-minion status
salt-minion is not running
Starting service salt-master
Checking service salt-master status
salt-master is running
Starting service salt-minion
Checking service salt-minion status
salt-minion is running
Checking if we can talk to the salt master
Can talk to salt master
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120003746491379
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120003803928388
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120003821326842
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120003838693667
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120003856009452
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120003913319687
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120003930637106
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120003947944252
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004005248619
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004022556362
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004039861594
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004057164727
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004114457169
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004131765837
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004149072954
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004206377281
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004223678529
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004240993020
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004258302393
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004315605144
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004332922359
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004350234695
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004407535971
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004424860189
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004442158015
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004459473595
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004516786075
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004534110808
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004551429983
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004608733013
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004626054688
salt master did not get a job response from salt minion
Checking if the salt minion will respond to jobs
ERROR: Minions returned with non-zero exit code
securityonion_eval:
    Minion did not return. [No response]
    The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:
    
    salt-run jobs.lookup_jid 20201120004643370892
salt master did not get a job response from salt minion
salt master did not get a job response from salt minion after 30 attempts, exiting
Errors detected during setup; skipping post-setup steps to allow for analysis of failures.
Installer removing the following files:
/root/installtmp:
total 4
drwxr-xr-x 3 root root 4096 Nov 20 00:34 pillar

/root/installtmp/pillar:
total 4
drwxr-xr-x 2 root root 4096 Nov 20 00:34 minions

/root/installtmp/pillar/minions:
total 4
-rw-r--r-- 1 root root 1048 Nov 20 00:37 securityonion_eval.sls
=========================================================================
Stopping mysql...
=========================================================================
Error response from daemon: No such container: so-mysql
Error: No such container: so-mysql

**Reply by Courts3003** [reply]
For me, I now have SO 2.3.2 working. My 4 platform distributed configuration is absolutely awesome. 

Hats of to the guys at Security Onion for such a comprehensive upgrade. Well done!

I was having issues with the Salt Infrastructure also but I managed to solve them for my complex network (Clustered Cisco 2900s, Multiple KVM virtualisation servers supporting genevre Layer 3 VLANs (i.e. a cloud)).

I found that for my network I could not run the Master-Sensors across different LAN segments. Salt simply would not install on the Sensor, it could not find the Salt Master. It looked like a network stack problem to me. I can nc -v any protocol successfully across LAN segments on my firewall apart from Salt TCP ports. As soon as I placed the Master and the Sensor in the same segment LAN segment (e.g. Management NIC of 192.168.x.x/24) it all worked. You can monitor different segments by attaching your sensor monitoring NIC to the different segment to be managed rather than deploying the complete Sensor (Management NIC and Monitoring NIC) on the separate LAN segment.

That's just what I found work for my environment.

**Comment by femaven** [comment, marked as answer]
My issue was resolved in SO 2.3.21.
